tests/storage/outofprocess_test.py

```diff
@@ -49,6 +50,20 @@ def oop_cleanup():
     oop.stop()
 
 
+@pytest.fixture(
+    scope="module",
+    params=[
+        userstorage.PATHS["file-512"],
+    ],
+    ids=str,
+)
+def user_file(request):
+    storage = request.param
+    if not storage.exists():
+        pytest.xfail("{} storage not available".format(storage.name))
+    return storage
```

Comment:
We need to reset this file before each test so tests cannot be affected by previous tests. In other modules using this storage we truncate the file before each test.

Example:
https://github.com/oVirt/vdsm/blob/feb336ae643e0342df74877819a3dec10ed10bb4/tests/storage/filesd_metadata_test.py#L114

But looking at the tests, we really want a temporary directory and not a file, so we should use "mount-512" instead.

The current tests use a temporary directory, so the easier and smallest change is to create a fixture returning a new temporary directory, and delete the temporary directory no the end of the test.

Example: https://github.com/oVirt/vdsm/blob/feb336ae643e0342df74877819a3dec10ed10bb4/tests/storage/localfssd_test.py#L78

Since the fixture return a path instead of pytedt tmpdir, the tests will have to change code like:

    path = str(tmpdir.join("file"))

To:

    path = os.path.join(user_tmpdir, "file")

But no other change is needed in the tests.

Reply:
OK.

I have used [pathlib](https://docs.python.org/3/library/pathlib.html) instead of `os.path`, since it is much nicer (in this case is not such a big difference, and one could argue that `os` was already imported in the module, but start introducing a library little by little is an approach, and `pathlib` is there since v3.4).

Also, I changed the fixture a bit. I checked `localfssd_test` as an example, but changed it differently. I have changed the scope to function, so that it is executed for every test:
- It creates the file Path from inside the fixture, to avoid code repetition
- Then yields it instead of the raw mounting point
- Finally, it removes the file so that each test has to create it anew every time

But I can just change it to follow the example as in `localfssd_test`, I was experimenting a bit :)